engine/orchestration/src/com/cloud/vm/VirtualMachineManagerImpl.java

```diff
@@ -1331,10 +1414,35 @@ public boolean getExecuteInSequence(final HypervisorType hypervisorType) {
         }
     }
 
+    private List<Map<String, String>> getVolumesToDisconnect(VirtualMachine vm) {
+        List<Map<String, String>> volumesToDisconnect = new ArrayList<>();
+
+        List<VolumeVO> volumes = _volsDao.findByInstance(vm.getId());
+
+        if (volumes != null) {
```

Comment:
Here you can also do the same, if volumes is empty, you can return right away the empty `volumesToDisconnect`, thus removing the next operations from a IF statement

Reply:
Same as above